diagnostic_feedback/tests/integration/base_test.py

```diff
@@ -0,0 +1,241 @@
+        """
+        return [c for c in self.browser.find_elements_by_xpath("//input[@type='radio']") if c.is_displayed()]
+
+    def _is_any_choice_selected(self):
+        """
+        Check if any choice is selected from visible choices
+        """
+        return any(choice.is_selected() for choice in self._get_visible_choices())
+
+    def _verify_back_button_status(self, back_btn, question_order):
+        """
+        Check if previous button is enabled for a question that has order > 0
+        """
+        if question_order > 0:
+            # if question is not first in order, previous button should be enabled
+            self.assertEqual(back_btn.get_attribute('class'), '')
+        else:
+            # if first question, previous button should be disabled
+            self.assertEqual(back_btn.get_attribute('class'), 'disabled')
+
+    def _verify_question(self, question_order=0, next_btn=None, back_btn=None, choice_idx=0):
+        # confirm that not a single choice is selected from visible choices as a question is initially loaded
+        self.assertFalse(self._is_any_choice_selected())
+
+        # check if previous button is properly enabled/disabled for any question, before a choice selection
+        self._verify_back_button_status(back_btn, question_order)
+
+        # next button should be disabled until a choice is selected
+        self.assertEqual(next_btn.get_attribute('class'), 'disabled')
+
+        # get a given choice and click to select
+        choice = self._get_visible_choices()[choice_idx]
+        choice.click()
+        # confirm if choice is marked as selected
+        self.assertTrue(self._is_any_choice_selected())
+
+        # check if previous button is properly enabled/disabled for any question, after a choice selection
+        self._verify_back_button_status(back_btn, question_order)
+
+        # check if next button is enabled after any choice selection
+        self.assertEqual(next_btn.get_attribute('class'), '')
+
+    def _verify_startover(self, next_btn, back_btn, next_btn_link, back_btn_link, startover_link):
+        """
+        Method to verify start-over functionality
+        """
+
+        # click start-over link, it will load first question of quiz
+        startover_link.click()
+        # wait until hidden next button is shown
+        self.wait_until_visible(next_btn)
+        # check if start-over button is hidden properly
+        self.assertFalse(startover_link.is_displayed())
+        # check if next/previous buttons have 'disabled' classed after shown
+        self.wait_until_has_class_disabled(next_btn)
+        self.assertEqual(back_btn.get_attribute('class'), 'disabled')
+
+        # verify 1st question for all necessary checks, also select 1st choice for this question
+        self._verify_question(question_order=0, next_btn=next_btn, back_btn=back_btn, choice_idx=0)
+        # submit 1st question, it should load next question i;e (question => 2)
```

Comment:
@jareerahsan What I meant to say was, please also remove the `i;e` part -- I'm not sure what it means? If you were going for [i.e.](https://en.wiktionary.org/wiki/i.e.), it would make sense to move it into the parenthesized portion of the comment:

```python
# submit 1st question, it should load next question (i.e., question 2)
```

Up to you if you want to include it or not; I think the comments are clear without it.